flox/xarray.py

```diff
@@ -194,6 +194,7 @@ def xarray_reduce(
     if skipna is not None and isinstance(func, Aggregation):
         raise ValueError("skipna must be None when func is an Aggregation.")
 
+    by_len = len(by)
```

Comment:
minor comment: can we call this `nby` instead. I use that in `core.py` I think.

Lets also add a test